### ISSUE TITLE
"loading" flag of layer source not set to false if loading is interrupted.

### ISSUE THREAD

**Issue by mktcode** [openlayers/openlayers#9503]
**The Problem**
- We want a loading animation when layer sources are loading/refreshed.
- `console.log(layer.getSource())` shows there is a `loading` flag.
- We have layers with a `maxResolution`.
- The map is initialized with a certain zoom/extent.
- After a HTTP request to fetch a user specific map extent the map is moved to that new extent.
- Thereby the `maxResolution` of these layers is exceeded and their loading process is interupted.
- The `loading` flag of the sources of these layers is still `true`.
- Thus our loading animation does not stop.

**To Reproduce**
- Configure a layer to have a `maxResolution`.
- Initialize a map with a zoom/extent so that this layer is actually displayed.
- `console.log(layer.getSource())`: `loading: true`
- Zoom out of the `maxResolution`  so that the layer is not displayed anymore, while the layer source is still loading.
- `console.log(layer.getSource())`: `loading: true` (stays like that)

**Expected behavior**
Since the loading process is interrupted I expect the loading flag to be set to `false` after that. Instead it stays `true`, giving the impression that the loading process is still ongoing.

**Comment by ahocevar** [comment]
That `loading` property is not part of the API. it is only used internally. If you want to create a loading indicator, you should be listening once to the map's `rendercomplete` event. Does that work in your scenario?

**Comment by mktcode** [comment]
`rendercomplete` is fired far too early. I also don't want to listen only once. I want to display a loading indicator whenever something on the map is loading. Be it by paning/zooming or refreshing layer sources manually or whatever. The `loading` flag of the layer sources behaves exactly as I want, except for the behavior outlined above. Is there any other reliable way to achive what I want? I saw that layer sources have a `getState` method. But that seems to return only `ready` although in the [docs](https://openlayers.org/en/latest/apidoc/module-ol_source_State.html) it mentions a `loading` state.

**Comment by mktcode** [comment]
@ahocevar Any idea what would be a proper way to implement that?

**Comment by Rskut** [comment]
#13491
